infra/azure/main.tf

```diff
@@ -0,0 +1,97 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "=2.70.0"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "1.13.3"
+    }
+  }
+  backend "azurerm" {}
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_resource_group" "rg" {
+  name = var.az_resource_group_name
+}
+
+resource "azurerm_virtual_network" "default" {
+  name = "default"
+  resource_group_name = data.azurerm_resource_group.rg.name
+  address_space = ["10.0.0.0/8"]
+  location = data.azurerm_resource_group.rg.location
+}
+
+resource "azurerm_subnet" "k8s_subnet" {
+  name                 = "k8s-subnet"
+  address_prefixes     = ["10.240.0.0/16"]
+  resource_group_name  = data.azurerm_resource_group.rg.name
+  virtual_network_name = azurerm_virtual_network.default.name
+}
+
+resource "azurerm_subnet" "batch_worker_subnet" {
+  name                 = "batch-worker-subnet"
+  address_prefixes     = ["10.128.0.0/16"]
+  resource_group_name  = data.azurerm_resource_group.rg.name
+  virtual_network_name = azurerm_virtual_network.default.name
+}
+
+resource "azurerm_kubernetes_cluster" "vdc" {
+  name                = "vdc"
+  resource_group_name = data.azurerm_resource_group.rg.name
+  location            = data.azurerm_resource_group.rg.location
+  dns_prefix          = "example"
+
+  default_node_pool {
+    name       = "default"
+    node_count = 1
+    vm_size    = "Standard_D2_v2"
+    vnet_subnet_id = azurerm_subnet.k8s_subnet.id
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_kubernetes_cluster_node_pool" "vdc_pool" {
+  name = "pool"
+  kubernetes_cluster_id = azurerm_kubernetes_cluster.vdc.id
+  vm_size = "Standard_D2_v2"
+  vnet_subnet_id = azurerm_subnet.k8s_subnet.id
+
+  enable_auto_scaling = true
+
+  node_count = 1
+  min_count = 0
+  max_count = 200
+}
+
+resource "random_id" "db_name_suffix" {
+  byte_length = 4
+}
+
+resource "random_password" "db_root_password" {
+  length = 22
+}
+
+resource "azurerm_mysql_server" "db" {
+  name                = "db-${random_id.db_name_suffix.hex}"
+  resource_group_name = data.azurerm_resource_group.rg.name
```

Comment:
yup! I think I need to understand the mysql connector on the python side first because I'm not sure if I can 1:1 match the existing GCP secret (or maybe I can and I haven't dug enough yet).